### PR TITLE
Add node condition for Microsoft.ContainerInstance provider registration and couple of fixes

### PR DESCRIPTION
**Pull request by kkmsft**
Add node condition for Microsoft.ContainerInstance provider registration
Dedup the condition generation code.
Fix the version string in the node status for the first update.

### CODE REVIEW

**Comment by lachie83** [comment]
cc @brendandburns

File: node.ts [hunk 1]
```diff
@@ -1,63 +1,97 @@
...
+            message: "kubelet has sufficient disk space available",
+            reason: "KubeletHasSufficientDisk",
+            status: "False",
+            type: "OutOfDisk"
+        } as api.V1NodeCondition
+    );
+
+    if (!provider_registered) {
+        let provider = await providers.getProvider(rsrcClient);
+        if (provider['registrationState'] != "Registered") {
```
Comment (brendandburns) [inline node.ts]: nit: single-quote for strings.

File: node.ts [hunk 2]
```diff
@@ -1,63 +1,97 @@
...
+            conditions.push(
+                {
+                    lastHeartbeatTime: new Date(),
+                    lastTransitionTime: last,
+                    message: "Microsoft.ContainerInstance not registered",
```
Comment (brendandburns) [inline node.ts]: Here too (and everywhere else)

File: node.ts [hunk 3]
```diff
@@ -1,63 +1,97 @@
...
+        } else {
+            // We will switch this flag once in a run..
+            provider_registered = true;
+        }
+    }
+    return conditions;
+}
+
+let updateNode = async (name: string, rsrcClient: azureResource.ResourceManagementClient, transition: Date, client: api.Core_v1Api, keepRunning: () => boolean) => {
     console.log("sending update.");
```
Comment (brendandburns) [inline node.ts]: clearly I'm not consistent though... :P

File: providers.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import azureResource = require('azure-arm-resource');
+import rest = require('ms-rest');
+
+export async function getProvider(client: azureResource.ResourceManagementClient): Promise<Object> {
```
Comment (brendandburns) [inline providers.ts]: Can we use the actual type instead of `Object`?

**Comment by brendandburns** [comment]
Thanks for the PR. Generally this looks great! some small comments.

File: node.ts (hunk 1 above)
Comment (kkmsft) [reply node.ts]: Fixed.

File: node.ts (hunk 2 above)
Comment (kkmsft) [reply node.ts]: Fixed in the latest  update to the PR.

File: node.ts (hunk 3 above)
Comment (kkmsft) [reply node.ts]: :-) 
Fixed this string. Fixed the ones which the PR is touching.

File: providers.ts (hunk 4 above)
Comment (kkmsft) [reply providers.ts]: Changed to the type as per review - azureResource.ResourceModels.Provider

**Comment by brendandburns** [comment]
LGTM, thanks again.

**Pull request merged**